### PR TITLE
Implement `blockchain.transaction.id_from_pos` RPC

### LINKED ISSUE #834: Feature: Implement `blockchain.transaction.id_from_pos` method
Implement `blockchain.transaction.id_from_pos` protocol method as described in the [specs here](https://electrumx-spesmilo.readthedocs.io/en/latest/protocol-methods.html#blockchain-transaction-id-from-pos).

I would like to work on this issue @romanz

### PR DESCRIPTION
**Pull request by EshaanAgg**
Add [blockchain.transaction.id_from_pos](https://electrumx-spesmilo.readthedocs.io/en/latest/protocol-methods.html#blockchain-transaction-id-from-pos) protocol method.

Fixes #834.

### CODE REVIEW

File: src/electrum.rs [hunk 1]
```diff
@@ -401,6 +401,35 @@ impl Rpc {
...
+        let chain = self.tracker.chain();
+        let blockhash = match chain.get_block_hash(*height) {
+            None => bail!("missing block at {}", height),
+            Some(blockhash) => blockhash,
+        };
+        let txids = self.daemon.get_block_txids(blockhash)?;
+        let txid: Txid = txids[*tx_pos];
+        if *merkle {
+            match txids.iter().position(|current_txid| *current_txid == txid) {
+                None => bail!("missing txid {} in block {}", txid, blockhash),
```
Comment (romanz) [inline src/electrum.rs:417]: Not sure that we may get here, since `txid == txids[*tx_pos]`, so `position == *tx_pos`...

Comment (EshaanAgg) [reply src/electrum.rs:420]: Added test for invalid `tx_pos` in the latest commit.

Comment (romanz) [reply src/electrum.rs]: Pushed a small simplification commit (https://github.com/romanz/electrs/pull/836/commits/5928b990442ee50f6df71af0db38b44fdb08ddd4) - please take a look :)

**Comment by EshaanAgg** [comment]
LGTM @romanz! The code is much easier to read now!

**Pull request merged**